files/hipchat.rb

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env ruby
+
+require "#{File.dirname(__FILE__)}/base"
+#require_relative 'base'
+
+class Hipchat < BaseHandler
+  def handle
+    timeout_and_retry do
+      case @event['check']['status'].to_i
+      when 1,2
+        trigger_incident
+      when 0
+        resolve_incident
+      end
+    end
+  end
+
+  def api_key
+    handler_settings['api_key'] || false
+  end
+
+  def trigger_incident
+    return false unless api_key
+    alert_hipchat(
+      hipchat_room,
+      'sensu',
+      hipchat_message,
+      { :color => hipchat_message_colour, :notify =>  true }
+    )
+  end
+
+  def resolve_incident
+    return false unless api_key
+    alert_hipchat(
+      hipchat_room,
+      'sensu',
+      hipchat_message,
+      { :color => hipchat_message_colour }
+    )
+  end
+
+  def event_time
+    Time.at(@event['check']['issued']).utc.strftime "%Y-%m-%d %H:%M:%S UTC"
+  end
+
+  def hipchat_message
+"
+<b>#{event_time} - #{@event['check']['name']} on #{@event['client']['name']} (#{@event['client']['address']}) - #{human_check_status}</b>
+<br /><br />
+&nbsp;&nbsp;#{@event['check']['notification'] || @event['check']['output']}
+"
+  end
+
+  def hipchat_message_colour
+    case @event['check']['status']
+    when 0
+      'green'
+    when 1
+      'yellow'
+    when 2
+      'red'
+    else
+      'grey'
+    end
+  end
+
+  def default_room
+    handler_settings['hipchat_room']
+  end
+
+  def hipchat_room
+    team_data('hipchat_room') || default_room || nil
```

Comment:
is the goal just to have the interface similar?   because currently default_room is still pulling from the handler_settings which is still specific to each handler, and therefore still configured independently.   

I think I only half understand where you're trying to get with this. 

 oh..  but at a higher level  we are still feeding the same team hash to all the per handler config  I think I see what you're getting at now. 

ok, yeah I think making the handlers respond to room or channel in the team hash could be a good pattern.  assuming I understand you correctly.